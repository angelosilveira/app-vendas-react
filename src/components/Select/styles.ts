import styled from 'styled-components';

export const Wrapper = styled.div`
  margin-bottom: 1rem;

  select {
    display: block;
    width: 100%;
    padding: 1rem 0.5rem;
    font-size: 1.2rem;
    font-weight: 600;
    line-height: 1.5;
    color: #212529;
    background-color: #fff;
    background-repeat: no-repeat;
    background-position: right 0.75rem center;
    background-size: 16px 12px;
    border: 1px solid #ced4da;
    border-radius: 0.25rem;
    transition: border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
  }
`;

export const Error = styled.div`
  color: #f00;
  font-weight: bold;
  margin-top: 0.25rem;
  font-size: 1.2rem;
`;
