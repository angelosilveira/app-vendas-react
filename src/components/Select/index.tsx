import { forwardRef, HTMLAttributes } from 'react';
import * as S from './styles';

type OptionsProps = {
  value: string;
  label: string;
};

export type SelectProps = {
  options: OptionsProps[];
  title: string;
  label?: string;
  error?: string;
} & HTMLAttributes<HTMLSelectElement>;

const Select = (
  { options, title, label, error, ...rest }: SelectProps,
  ref: any
) => (
  <S.Wrapper>
    {!!label && <label>{label}</label>}
    <select {...rest} ref={ref}>
      <option value="" disabled>
        {title}
      </option>
      {options.map((option) => (
        <option key={option.value} value={option.value}>
          {option.label}
        </option>
      ))}
    </select>
    {!!error && <S.Error>{error}</S.Error>}
  </S.Wrapper>
);

export default forwardRef(Select);
