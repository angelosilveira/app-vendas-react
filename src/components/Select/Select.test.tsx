import { screen } from '@testing-library/react';
import { renderWithTheme } from 'src/utils/tests/renderWithTheme';

import Select from '.';

const arrMock = [
  { label: 'Acre', value: 'AC' },
  { label: 'Alagoas', value: 'AL' },
  { label: 'Amap√°', value: 'AP' },
];

describe('<Select />', () => {
  it('should render the select', () => {
    const { container } = renderWithTheme(
      <Select title="Estados" options={arrMock} />
    );

    expect(screen.getByText('Estados')).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
