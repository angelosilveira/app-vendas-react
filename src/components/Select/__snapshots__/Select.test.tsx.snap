// Vitest Snapshot v1

exports[`<Select /> > should render the select 1`] = `
.c0 {
  margin-bottom: 1rem;
}

.c0 select {
  display: block;
  width: 100%;
  padding: 1rem 0.5rem;
  font-size: 1.2rem;
  font-weight: 600;
  line-height: 1.5;
  color: #212529;
  background-color: #fff;
  background-repeat: no-repeat;
  background-position: right 0.75rem center;
  background-size: 16px 12px;
  border: 1px solid #ced4da;
  border-radius: 0.25rem;
  -webkit-transition: border-color 0.15s ease-in-out,box-shadow 0.15s ease-in-out;
  transition: border-color 0.15s ease-in-out,box-shadow 0.15s ease-in-out;
}

<div
  class="c0"
>
  <select>
    <option
      disabled=""
      value=""
    >
      Estados
    </option>
    <option
      value="AC"
    >
      Acre
    </option>
    <option
      value="AL"
    >
      Alagoas
    </option>
    <option
      value="AP"
    >
      Amap√°
    </option>
  </select>
</div>
`;
